frontend/src/metabase/admin/settings/components/widgets/CustomGeoJSONWidget.jsx

```diff
@@ -0,0 +1,307 @@
+import React, { Component, PropTypes } from "react";
+
+import Utils from "metabase/lib/utils";
+
+import Select, { Option } from "metabase/components/Select.jsx";
+import Confirm from "metabase/components/Confirm.jsx";
+import Ellipsified from "metabase/components/Ellipsified.jsx";
+import Modal from "metabase/components/Modal.jsx";
+import LoadingAndErrorWrapper from "metabase/components/LoadingAndErrorWrapper.jsx";
+
+import SettingHeader from "../SettingHeader.jsx";
+
+import cx from "classnames";
+import fetch from 'isomorphic-fetch';
+
+import LeafletChoropleth from "metabase/visualizations/components/LeafletChoropleth.jsx";
+
+import pure from "recompose/pure";
+
+export default class CustomGeoJSONWidget extends Component {
+    constructor(props, context) {
+        super(props, context);
+        this.state = {
+            map: null,
+            originalMap: null,
+            geoJson: null,
+            geoJsonLoading: false,
+            geoJsonError: null,
+        };
+    }
+
+    static propTypes = {
+        setting: PropTypes.object.isRequired,
+        updateSetting: PropTypes.func.isRequired,
+        reloadSettings: PropTypes.func.isRequired
+    };
+    static defaultProps = {};
+
+    _saveMap = async (id, map) => {
```

Comment:
The `_` prefix usually indicates a "private" method (only called from within the class), but you're right we're pretty inconsistent. In React you don't really call any methods on components from outside except for the lifecycle methods which are called by React itself, but I like including the underscore for consistency with non-React code, and as a little differentiator from the lifecycle methods.

This is the sort of thing we should write out in a code style guide.